scripts/bootstrap_bench_dual_source.py

```
@@ -0,0 +1,177 @@
+def _fetch(method: str, base: str, path: str, token: str | None = None, body: dict | None = None) -> tuple[int, dict]:
+    url = f"{base.rstrip('/')}{path}"
+    headers: dict[str, str] = {}
+    if token:
+        headers["Authorization"] = f"Bearer {token}"
+    data = None
+    if body is not None:
+        headers["Content-Type"] = "application/json"
+        data = json.dumps(body).encode("utf-8")
+    req = urllib.request.Request(url, data=data, headers=headers, method=method)
+    try:
+        with urllib.request.urlopen(req, timeout=60) as resp:
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)